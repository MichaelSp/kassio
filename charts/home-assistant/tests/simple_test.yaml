suite: Simple Test Suite
templates:
- configmap.yaml
- deployment.yaml
- service.yaml
- serviceaccount.yaml
chart:
  version: 0.1.0
  appVersion: 1.2.3
tests:
- it: Matches the snapshot
  asserts:
  - matchSnapshot: {}
- it: has config
  set:
    persistence:
      enabled: true
      config:
        enabled: true
  asserts:
  - matchSnapshot: {}
- it: has extraConfigYaml
  set:
    configuration:
      extraConfigYaml: |
        some: thing
        other:
          complex:
            stuff
  asserts:
  - matchSnapshot: {}
- it: has /dev/ttyUSB0
  set:
    volumes:
    - name: ttyUSB0
      hostPath:
        path: /dev/ttyUSB0
        type: CharDevice
    volumeMounts:
    - name: ttyUSB0
      mountPath: /dev/ttyUSB0
  asserts:
  - matchSnapshot: {}
- it: has postgresql.enabled
  set:
    postgresql:
      enabled: true
  asserts:
  - matchSnapshot: {}
