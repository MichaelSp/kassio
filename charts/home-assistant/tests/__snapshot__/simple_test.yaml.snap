Matches the snapshot:
  1: |
    apiVersion: v1
    data:
      configuration.yaml: |
        # Loads default set of integrations. Do not remove.
        default_config:
        # Load frontend themes from the themes folder
        frontend:
          themes: !include_dir_merge_named themes

        automation: !include automations.yaml
        script: !include scripts.yaml
        scene: !include scenes.yaml
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-home-assistant-config
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: home-assistant
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: home-assistant
            app.kubernetes.io/version: 1.2.3
            helm.sh/chart: home-assistant-0.1.0
        spec:
          containers:
            - image: ghcr.io/home-assistant/home-assistant:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: home-assistant
              ports:
                - containerPort: 8123
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /config
                  name: home-assistant-config
          securityContext: {}
          serviceAccountName: RELEASE-NAME-home-assistant
          volumes:
            - configMap:
                name: RELEASE-NAME-home-assistant-config
              name: home-assistant-config
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      ports:
        - name: http
          port: 8123
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: home-assistant
      type: ClusterIP
  4: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
has /dev/ttyUSB0:
  1: |
    apiVersion: v1
    data:
      configuration.yaml: |
        # Loads default set of integrations. Do not remove.
        default_config:
        # Load frontend themes from the themes folder
        frontend:
          themes: !include_dir_merge_named themes

        automation: !include automations.yaml
        script: !include scripts.yaml
        scene: !include scenes.yaml
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-home-assistant-config
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: home-assistant
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: home-assistant
            app.kubernetes.io/version: 1.2.3
            helm.sh/chart: home-assistant-0.1.0
        spec:
          containers:
            - image: ghcr.io/home-assistant/home-assistant:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: home-assistant
              ports:
                - containerPort: 8123
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /config
                  name: home-assistant-config
                - mountPath: /dev/ttyUSB0
                  name: ttyUSB0
          securityContext: {}
          serviceAccountName: RELEASE-NAME-home-assistant
          volumes:
            - configMap:
                name: RELEASE-NAME-home-assistant-config
              name: home-assistant-config
            - hostPath:
                path: /dev/ttyUSB0
                type: CharDevice
              name: ttyUSB0
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      ports:
        - name: http
          port: 8123
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: home-assistant
      type: ClusterIP
  4: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
has config:
  1: |
    apiVersion: v1
    data:
      configuration.yaml: |
        # Loads default set of integrations. Do not remove.
        default_config:
        # Load frontend themes from the themes folder
        frontend:
          themes: !include_dir_merge_named themes

        automation: !include automations.yaml
        script: !include scripts.yaml
        scene: !include scenes.yaml
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-home-assistant-config
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: home-assistant
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: home-assistant
            app.kubernetes.io/version: 1.2.3
            helm.sh/chart: home-assistant-0.1.0
        spec:
          containers:
            - image: ghcr.io/home-assistant/home-assistant:1.2.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
              name: home-assistant
              ports:
                - containerPort: 8123
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: http
              resources: {}
              securityContext: {}
              volumeMounts:
                - mountPath: /config
                  name: home-assistant-config
          securityContext: {}
          serviceAccountName: RELEASE-NAME-home-assistant
          volumes:
            - configMap:
                name: RELEASE-NAME-home-assistant-config
              name: home-assistant-config
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
    spec:
      ports:
        - name: http
          port: 8123
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: home-assistant
      type: ClusterIP
  4: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: home-assistant
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: home-assistant-0.1.0
      name: RELEASE-NAME-home-assistant
